src/cmd/check.go

```diff
@@ -54,22 +56,74 @@ var listCheckCmd = &cobra.Command{
 var checkCreateCmd = &cobra.Command{
 	Use:   "check",
 	Short: "Create a rubric check",
-	Long: `Create a rubric check
-
-Examples:
-
-	opslevel create check -f my_cec.yaml
-`,
+	Long:  `Create a rubric check`,
+	Example: `
+cat << EOF | opslevel create check -f -
+Version: "1"
+kind: "repo_grep"
+spec:
+  name: "new repo grep check"
+  enabled: false
+  category: "misc"
+  level: "silver"
+  notes: "some notes"
+  directorySearch: false
+  filePaths:
+  - "Taskfile.yml"
+  fileContentsPredicate:
+    type: "exists"
+    value: ""
+EOF`,
 	Run: func(cmd *cobra.Command, args []string) {
-		usePrompts := !hasStdin()
-		input, err := readCheckCreateInput()
+		input, err := readCheckInput()
 		cobra.CheckErr(err)
+		usePrompts := true
+		if cmd.Flag("noninteractive").Changed {
+			usePrompts = false
+		} else if dataFile == "-" {
+			log.Warn().Msg("running noninteractively since using heredoc")
+			usePrompts = false
+		}
 		check, err := createCheck(*input, usePrompts)
 		cobra.CheckErr(err)
 		fmt.Printf("Created Check '%s' with id '%s'\n", check.Name, check.Id)
 	},
 }
 
+var checkUpdateCmd = &cobra.Command{
+	Use:   "check",
+	Short: "Update a rubric check",
+	Long:  `Update a rubric check`,
+	Example: `
+cat << EOF | opslevel update check $CHECK_ID -f -
+Version: "1"
+kind: "repo_grep"
+spec:
+  name: "updated repo grep check"
+  enabled: true
+  fileContentsPredicate:
+    type: "does_not_match_regex"
+    value: "gofmt"
+EOF`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID"},
+	Run: func(cmd *cobra.Command, args []string) {
+		input, err := readCheckInput()
+		cobra.CheckErr(err)
+		input.Spec["id"] = *opslevel.NewID(args[0])
```

Comment:
The actual mutation inputs used by opslevel-go are created by inserting all the key-value pairs from `Spec` into a graphql request. 

We can use the same input type for `create` and `update` by simply adding the ID here, similar to how we are working with filters. https://github.com/OpsLevel/cli/blob/7a230568bc3dd94545c8b3cb57be0b610c06f2b2/src/cmd/filter.go#L99-L107

Reply:
Ya this is legacy from before Evan joined and showed us the way to make input types not unique between create and update.